src/test/java/com/wikia/webdriver/elements/oasis/components/templateclassificiation/TemplateClassification.java

```diff
@@ -0,0 +1,105 @@
+package com.wikia.webdriver.elements.oasis.components.templateclassificiation;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.core.elemnt.Wait;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+public class TemplateClassification {
+
+  @FindBy(css = ".template-classification-type-label")
+  private WebElement entryPointLink;
+
+  @FindBy(css = "#template-classification-infobox")
+  private WebElement typeInputInfobox;
+
+  @FindBy(css = "#template-classification-quote")
+  private WebElement typeInputQuote;
+
+  @FindBy(css = ".button.normal.primary")
+  private WebElement saveButton;
+
+  @FindBy(css = ".template-classification-edit-modal .close")
+  private WebElement closeButton;
+
+  private By modalSelector = By.cssSelector(".template-classification-edit-modal");
+
+  private String templateName = "";
+
+  private Wait wait;
+
+  public TemplateClassification(WebDriver driver) {
+    this.wait = new Wait(driver);
+
+    PageFactory.initElements(driver, this);
+  }
+
+  public TemplateClassification open() {
+    wait.forElementClickable(entryPointLink);
+    this.templateName = entryPointLink.getText();
+    entryPointLink.click();
+
+    wait.forElementVisible(modalSelector);
+    PageObjectLogging.logInfo("Template classification modal was opened");
+
+    return this;
+  }
+
+  public TemplateClassification close() {
+    wait.forElementClickable(closeButton);
+    closeButton.click();
+
+    wait.forElementNotVisible(modalSelector);
+    PageObjectLogging.logInfo("Template classification modal was closed");
+
+    return this;
+  }
+
+  public TemplateClassification save() {
+    String oldTemplateName = this.templateName;
+    wait.forElementClickable(saveButton);
+    saveButton.click();
+
+    wait.forElementVisible(entryPointLink);
+    String currentTemplateName = entryPointLink.getText();
+    Assertion.assertFalse(currentTemplateName.equals(oldTemplateName),
+                          "Template type did not change");
+    PageObjectLogging.logInfo("Template type changed from: '" + oldTemplateName +
+                              "', to: '" + currentTemplateName + "'");
+
+    return this;
+  }
+
+  public TemplateClassification selectTemplateType() {
+    if (!this.templateName.equals("Infobox")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "Infobox" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1132)